lib/ui/painting.dart

```diff
@@ -1632,6 +1632,19 @@ class Image {
     }
   }
 
+  /// Whether this reference to the underlying image is [dispose]d.
+  ///
+  /// This only returns a valid value if asserts are enabled, and must not be
+  /// used otherwise.
+  bool get debugDisposed {
+    bool? disposed;
+    assert(() {
+      disposed = _disposed;
+      return true;
+    }());
+    return disposed ?? (throw StateError('Image.debugDisposed is only available when asserts are enabled.'));
```

Comment:
On the web side we have a utility getter `assertionsEnabled`. If we add one in mobile, it could clean up the code here a bit:

```dart
if (assertionsEnabled) {
  return _disposed;
}
throw StateError(...);
```

Reply:
Done.

How is this set? A Dart VM define?

Reply:
Is this done here? However it's set, it will have to be a constant so the debug-only code gets DCEd.

Reply:
This will not get DCE'd if someone calls it in release mode, it will throw an exception. But the point is that people shouldn't be calling it in release mode.

Reply:
If someone says:
```dart
if (assertionsEnabled) {
  ... image.debugDisposed ...
}
```
and `assertionsEnabled` is a constant that is `false` in release, then the `if () {}` should be DCE'd, and if there are no calls `debugDisposed` can be tree-shaken.

Reply:
Wouldn't debugDisposed be tree-shaken either way if there are no calls?

Reply:
I'm not sure what you mean by "either way". If there is a call, and the call is guarded by `if (assertionsEnabled)` where `assertionsEnabled` is a compile-time constant, then the `if` block will be eliminated, and `debugDisposed` will be tree-shaken. And that's an if-and-only-if. If `assertionsEnabled` is not a compile-time constant, then the `if` block will not be eliminated, and `debugDisposed` will not be tree-shaken.

Reply:
Ahh ok - I think you and I are asking @yjbanov the same question then (is `assertionsEnabled` a compile time constant). I can't seem to find where it's defined.